### PR TITLE
Change preprocess parameters from GstBuffers to GstVideoFrame

### PR DESCRIPTION
**Pull request by GFallasRR**

### CODE REVIEW

File: ext/r2inference/gstgooglenet.c [hunk 1]
```diff
@@ -188,23 +196,36 @@ gst_googlenet_finalize (GObject * object)
...
-  gst_buffer_map (inbuf, &ininfo, GST_MAP_READ);
-  gst_buffer_map (outbuf, &outinfo, GST_MAP_WRITE);
+  static float adjusted[150528];
 
-  memcpy (outinfo.data, ininfo.data, ininfo.size);
+  for (int i = 0; i < scaled_size; i += channels) {
+    // BGR = RGB - Mean
+    //adjusted[i + 0] = (input[i + 2]);// - mean[0];
+    //adjusted[i + 1] = (input[i + 1]);// - mean[1];
+    //adjusted[i + 2] = (input[i + 0]);// - mean[2];
```
Comment (migueltaylor) [inline ext/r2inference/gstgooglenet.c]: Why is this commented?

File: ext/r2inference/gstgooglenet.c [hunk 2]
```diff
@@ -188,23 +196,36 @@ gst_googlenet_finalize (GObject * object)
...
+
+static gboolean
+gst_googlenet_preprocess (GstVideoInference * vi,
+    GstVideoFrame * inframe, GstVideoFrame * outframe)
+{
+  float* ret;
+
+  GST_LOG_OBJECT (vi, "Preprocess");
+  
+  ret = PreProcessImage(inframe->data[0], DIM, DIM, GoogleNetMean);
```
Comment (michaelgruner) [inline ext/r2inference/gstgooglenet.c]: Don't use DIM here, lets take the `width` and `height` from `inframe->info.width` and `inframe->info.height`

File: ext/r2inference/gstgooglenet.c [hunk 3]
```diff
@@ -188,23 +196,36 @@ gst_googlenet_finalize (GObject * object)
...
+  memcpy (outframe->data[0], ret, 602112);
```
Comment (michaelgruner) [inline ext/r2inference/gstgooglenet.c]: Don't use `602112` hardcoded number. Get the size from the info struct.

File: ext/r2inference/gstgooglenet.c [hunk 4]
```diff
@@ -55,12 +60,15 @@ static void gst_googlenet_dispose (GObject * object);
...
 
 static gboolean gst_googlenet_preprocess (GstVideoInference * vi,
-    GstBuffer * inbuf, GstBuffer * outbuf);
+    GstVideoFrame * inframe, GstVideoFrame * outframe);
 static gboolean gst_googlenet_postprocess (GstVideoInference * vi,
     GstBuffer * buf, const gpointer prediction, gsize predsize);
 static gboolean gst_googlenet_start (GstVideoInference * vi);
 static gboolean gst_googlenet_stop (GstVideoInference * vi);
 
+static float* PreProcessImage (const unsigned char *input, int reqwidth,
```
Comment (michaelgruner) [inline ext/r2inference/gstgooglenet.c]: the method naming convention is `pre_process_image`, not `PreProcessImage`

File: ext/r2inference/gstgooglenet.c (hunk 3 above)
Comment (michaelgruner) [inline ext/r2inference/gstgooglenet.c]: Why are you copying? Write from `input` to `output`

**Pull request closed without merging**